src/js/form-builder.js

```diff
@@ -108,27 +106,37 @@ const FormBuilder = function(opts, element) {
     // build & insert the new list item to represent this control
     let newFieldControl = m('li',
       m('span', label),
-      {className: `${iconClassName} input-control input-control-${controlIndex}`}
+      {className: `${iconClassName} input-control input-control-${i}`}
     );
     newFieldControl.dataset.type = type;
-    d.controls.appendChild(newFieldControl);
-
-    // map for later access
-    controlIndex++;
+    controlList.push(type);
+    allControls[type] = newFieldControl;
   }
 
   if (opts.inputSets.length) {
-    $('<li/>', {'class': 'fb-separator'}).html('<hr>').appendTo($cbUL);
     opts.inputSets.forEach((set, i) => {
       set.name = set.name || utils.makeClassName(set.label);
       let inputSet = m('li', set.label, {
-        className: `input-set-control input-set-${i}`,
-        type: set.name
+        className: `input-set-control input-set-${i}`
       });
-      $(inputSet).appendTo($cbUL);
+      inputSet.dataset.type = set.name;
+      controlList.push(set.name);
+      allControls[set.name] = inputSet;
     });
   }
 
+  // remove disableFields
+  if (opts.disableFields) {
+    controls = controls.filter(type => opts.disableFields.indexOf(type) == -1);
```

Comment:
this should go either before `for` loop above, or filter out controlList /  allControls